superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/customControls.tsx

```diff
@@ -66,10 +67,16 @@ function isForcedCategorical(controls: ControlStateMapping): boolean {
 }
 
 function isSortable(controls: ControlStateMapping): boolean {
+  const xAxisValue = controls?.x_axis?.value as QueryFormColumn;
+  // Given that we don't know the type of a custom SQL column,
+  // we treat it as sortable and give the responsibility to the
+  // user to provide a sortable result.
+  const isCustomSQL = !isPhysicalColumn(xAxisValue);
   return (
     isForcedCategorical(controls) ||
+    isCustomSQL ||
     checkColumnType(
-      getColumnLabel(controls?.x_axis?.value as QueryFormColumn),
+      getColumnLabel(xAxisValue),
```

Comment:
I was wondering why this was passing our 100 % coverage check, but it turns out this file is `tsx` which is ignored in the check. I'm thinking at some point we should really move this type of critical non-React logic into `.ts` files and start adding tests for them. For this specific function, I'd maybe move it under `src/shared-controls/utils/isSortable.ts` and then add a test for it.

Reply:
So, I tried to add a test for our shared controls but quickly find out that the way they are structured makes the tests really hard. I hate this weird declarative structure that we have instead of working with real React composition. I think we should postpone writing tests for them right now because we should rewrite them completely.

Reply:
@villebro I extracted the function and added tests.

Reply:
Niiice! ❤️